File: src/main/java/io/github/chaosdave34/kitpvp/commands/AddExperienceCommand.java
package io.github.chaosdave34.kitpvp.commands;

import io.github.chaosdave34.kitpvp.ExtendedPlayer;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class AddExperienceCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (args.length != 2) return false;

        Player player = Bukkit.getPlayer(args[0]);
        if (player == null) return false;

        int amount;
        try {
            amount = Integer.parseInt(args[1]);
        } catch (NumberFormatException e) {
            return false;
        }

        if (amount < 1) return false;

        ExtendedPlayer.from(player).addExperiencePoints(amount);

        return true;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/commands/AddExperienceCommand.kt
package io.github.chaosdave34.kitpvp.commands

import com.mojang.brigadier.Command
import com.mojang.brigadier.arguments.IntegerArgumentType
import io.github.chaosdave34.kitpvp.ExtendedPlayer
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import io.papermc.paper.command.brigadier.argument.ArgumentTypes
import io.papermc.paper.command.brigadier.argument.resolvers.selector.PlayerSelectorArgumentResolver
import net.minecraft.commands.arguments.EntityArgument

object AddExperienceCommand {
    fun register(commands: Commands) {
        commands.register(
            Commands.literal("add-experience")
                .requires { it.sender.isOp }
                .then(
                    Commands.argument("targets", ArgumentTypes.players())
                        .then(
                            Commands.argument("amount", IntegerArgumentType.integer(1))
                                .executes {
                                    addExperience(
                                        it.source,
                                        it.getArgument("targets", PlayerSelectorArgumentResolver::class.java),
                                        it.getArgument("amount", Int::class.java)
                                    )
                                }
                        )
                )
                .build()
        )
    }

    private fun addExperience(source: CommandSourceStack, targets: PlayerSelectorArgumentResolver, amount: Int): Int {
        val resolvedTargets = targets.resolve(source)
        if (resolvedTargets.isEmpty()) throw EntityArgument.NO_PLAYERS_FOUND.create()

        resolvedTargets.forEach { target ->
            ExtendedPlayer.from(target).addExperiencePoints(amount)
        }
        return Command.SINGLE_SUCCESS
    }

}